File: src/week4_1.java
import java.util.Scanner;

public class week4_1 {

    public static class Node {
        public Node next;
        public int key;

        public Node(int key){
            next = null;
            this.key = key;
        }  

    }


    public static class LinkedList {
        private Node head;
        private Node tail;

        public LinkedList(){
            this.head = null;
            this.tail = null;
        }

        public void pushFront(int key) {
            Node n = new Node(key);


            if(this.head == null) {
                this.head = n;
                this.tail = n;
            } else {
                n.next = this.head;
                this.head = n;
            }
        }

        public void pushBack(int key) {
            Node n = new Node(key);

            if(this.head == null) {
                this.head = n;
                this.tail = n;
            } else {
                this.tail.next = n;
                this.tail = n;
            }

        }

        public int pop(){
            if(this.head == null){
                System.out.println(0);
            }
            int res = this.head.key;
            if(this.tail == this.head){
                this.tail = null;
                this.head = null;
            } else {
                this.head = this.head.next;
            }
            return res;
        }
        
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int N = input.nextInt();
        LinkedList l = new LinkedList();

        for(int i = 0; i<N; i++) {
            String op = input.next();
            if(op.equals("PU")){
                l.pushFront(input.nextInt());
            } else {
                System.out.println(l.pop());
            }
        }

        input.close();
        
    }


}

File: Python_code/week4_1.py

class Node:
   def __init__(self, key):
      self.next = None
      self.key = key

class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None
    
    def push_front(self,key):
        n = Node(key)
        if self.head is None:
            self.head = n
            self.tail = n
        else:
            n.next = self.head
            self.head = n

    def push_back(self,key):
        n = Node(key)
        if self.tail is None:
            self.head = n
            self.tail = n
        else:
            self.tail.next = n
            self.tail = n

    def pop(self):
        res = self.head.key
        if(self.head == self.tail):
            self.head = None
            self.tail = None
        else:
            self.head = self.head.next
        return res


n = int(input())

l = LinkedList()

for _ in range(n):
    inp = input().split()
    if inp[0] == "PU":
        l.push_front(int(inp[1]))
    else:
        print(l.pop())


# brug push_back i stedet for push_front når man skal lave en queue